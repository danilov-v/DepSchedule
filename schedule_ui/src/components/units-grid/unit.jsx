import React from "react";
import PropTypes from "prop-types";
import { Row, Col } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import classnames from "classnames";

export function Unit({
  title,
  unitId,
  lastGen,
  lastOfGroup,
  onAddUnit,
  onEditUnit,
  onRemoveUnit,
}) {
  const addUnit = () => onAddUnit(unitId);
  const editUnit = () => onEditUnit({ unitId, title });
  const removeUnit = () => onRemoveUnit(unitId);
  const iconClass =
    "unit-action d-flex align-items-center justify-content-center h-100";
  return (
    <div
      className={classnames(
        "unit",
        "d-flex",
        "justify-content-center",
        "align-items-center",
        "text-center",
        { "unit-last-generation": lastGen },
        { "unit-last-of-group": lastOfGroup }
      )}
      data-unit-id={unitId}
    >
      {title}
      <Row className="unit-actions" noGutters>
        <Col xs={4} className={iconClass} onClick={addUnit}>
          <FontAwesomeIcon icon="plus" />
        </Col>
        <Col xs={4} className={iconClass} onClick={editUnit}>
          <FontAwesomeIcon icon="edit" />
        </Col>
        <Col xs={4} className={iconClass} onClick={removeUnit}>
          <FontAwesomeIcon icon="trash-alt" />
        </Col>
      </Row>
    </div>
  );
}

Unit.propTypes = {
  title: PropTypes.string.isRequired,
  unitId: PropTypes.number.isRequired,
  lastGen: PropTypes.bool,
};
