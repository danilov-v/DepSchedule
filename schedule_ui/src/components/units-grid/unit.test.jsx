import React from "react";
import { shallow } from "enzyme";

import { Unit } from "./unit";

describe("units-grid.jsx", () => {
  describe("<Unit />", () => {
    it("should render and match snapshot", () => {
      const component = shallow(<Unit title="title" />);

      expect(component).toMatchSnapshot();
    });

    test("should render and match snapshot with lastGen", () => {
      const component = shallow(<Unit title="title" lastGen />);

      expect(component).toMatchSnapshot();
    });
  });
});
