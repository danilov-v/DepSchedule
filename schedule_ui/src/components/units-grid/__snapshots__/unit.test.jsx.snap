// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`units-grid.jsx <Unit /> should render and match snapshot 1`] = `
<div
  className="unit d-flex justify-content-center align-items-center text-center"
>
  title
</div>
`;

exports[`units-grid.jsx <Unit /> should render and match snapshot with lastGen 1`] = `
<div
  className="unit d-flex justify-content-center align-items-center text-center unit-last-generation"
>
  title
</div>
`;
